src/mokernels/slfm.jl

```diff
@@ -0,0 +1,63 @@
+@doc raw"""
+    LatentFactorMOKernel(
+        g::AbstractVector{<:Kernel}, 
+        e::AbstractVector{<:Kernel}, 
+        A::AbstractMatrix
+        )
+
+A semiparametric kernel for problems involving multiple response variables.
+
+``k((x, p), (y, p)) = k_p(x, y) = \Sum^{Q}_{q=1} A_{pq}g_q(x, y) + e_p(x, y)``
+
+# Arguments
+- `g::AbstractVector{<:Kernel}`: an array of kernels
+- `e::AbstractVector{<:Kernel}`: an array of kernels
+- `A::AbstractMatrix`: an matrix of weights for the kernels of shape (length(e), length(g))
+
+
+# Reference:
+- [Seeger, Teh, and Jordan (2005)](https://infoscience.epfl.ch/record/161465/files/slfm-long.pdf)
+
+"""
+struct LatentFactorMOKernel{Tg, Te, TA <: AbstractMatrix} <: Kernel
+    g::Tg
+    e::Te
+    A::TA
+    function LatentFactorMOKernel(g, e, A::AbstractMatrix)
+        all(isa.(g, Kernel)) || error("`g` should be an collection of kernels")
+        all(isa.(e, Kernel)) || error("`e` should be an collection of kernels")
```

Comment:
Maybe get rid of the allocations here by writing
```suggestion
        all(gi isa Kernel for gi in g) || error("`g` should be an collection of kernels")
        all(ei isa Kernel for ei in e) || error("`e` should be an collection of kernels")
```
here?